java/src/main/java/com/powsybl/python/network/NetworkUtil.java

```diff
@@ -246,4 +247,37 @@ public static String getRegulatedElementId(Supplier<Terminal> regulatingTerminal
         Terminal terminal = regulatingTerminalGetter.get();
         return terminal.getConnectable() != null ? terminal.getConnectable().getId() : null;
     }
+
+    /**
+     * @param b bus in Bus/Breaker view
+     * @return bus in bus view containing b if there is one, or null if none.
+     */
+    public static Bus getBusViewBus(Bus b) {
+        // First we try the fast and easy way using connected terminals. Works for the vast majority of buses.
+        Bus busInBusView = b.getConnectedTerminalStream().map(t -> t.getBusView().getBus())
+                .filter(Objects::nonNull)
+                .findFirst()
+                .orElse(null);
+        if (busInBusView != null) {
+            return busInBusView;
+        }
+        // Didn't find using connected terminals. There is the possibility that the bus has zero connected terminal
+        // on its own but is still part of a Merged Bus via a closed retained switch. We examine this case below.
+        VoltageLevel voltageLevel = b.getVoltageLevel();
+        if (voltageLevel.getTopologyKind() == TopologyKind.BUS_BREAKER) {
+            // Bus/Breaker. There is an easy method directly available.
+            return voltageLevel.getBusView().getMergedBus(b.getId());
+        } else {
+            // Node/Breaker. We should probably build something more efficient on powsybl-core side to avoid having
```

Comment:
Indeed! even though this is a corner case which should rarely occur.
Besides each call to `getBusStreamFromBusViewBusId` loops over all the nodes of the voltage level graph.